src/vulkan-renderer/camera.cpp

```diff
@@ -28,8 +28,9 @@ void Camera::update_vectors() {
 
 void Camera::update_matrices() {
     if (m_type == CameraType::LOOK_AT) {
+        const float vertical_fov = 2.0f * glm::atan(glm::tan(glm::radians(m_fov) / 2.0f) / m_aspect_ratio);
```

Comment:
When does this gets called? Maybe we can cache that value?

Reply:
It gets called whenever `m_update_needed == true`, so whenever there's mouse movement. But we only really need to recreate the projection matrix if there's zooming or window resizing so we can probably cache it somehow yes.

Reply:
I added a cache for all individual members.